Projects/Embark/Sources/Viewables/Actions/Action.swift

```diff
@@ -22,11 +22,26 @@ struct ActionResponseData {
 
 extension Action: Viewable {
     func materialize(events _: ViewableEvents) -> (UIView, Signal<GraphQL.EmbarkLinkFragment>) {
-        let view = UIStackView()
-        view.axis = .vertical
-
         let bag = DisposeBag()
 
+        let outerContainer = UIStackView()
+        outerContainer.axis = .vertical
+        outerContainer.alignment = .center
+        
+        let widthContainer = UIStackView()
+        widthContainer.axis = .horizontal
+        outerContainer.addArrangedSubview(widthContainer)
+        
+        bag += outerContainer.didLayoutSignal.onValue{ _ in
+            widthContainer.snp.remakeConstraints { make in
+                make.width.equalTo(outerContainer.frame.width > 600 ? 600 : outerContainer.frame.width)
```

Comment:
is this for ipad?

Reply:
yep! was battling this for an insane while and this is the best solution I came up with 😒

Reply:
What about `UIDevice.current.userInterfaceIdiom == .pad` instead just so its clear what this is for?